### PR TITLE
Clean up docs and packages

### PR DESCRIPTION
**Pull request by enolfc**
# Summary 

Try to better describe how this works and fix packages that are broken after latests changes.

----

**Related issue :** #36

### CODE REVIEW

**Comment by enolfc** [comment]
No idea on why codacy is not reporting back results :(

**Comment by enolfc** [comment]
> No idea on why codacy is not reporting back results :(

Commenting seems to bring them :)

File: README.md [hunk 1]
```diff
@@ -5,41 +5,46 @@
...
+INDIGO Configuration Management Database (CMDB) (Using specific templates).
 
-The generated representation is described using a
-[Mako](http://www.makotemplates.org/) template having access to the cloud
-middleware information.
+The provider extracts information from a cloud deployment using its public API
+and formats it according to [Mako](http://www.makotemplates.org/) templates.
 
-Supported cloud middleware providers:
+Current supported cloud middleware:
```
Comment (gwarf) [inline README.md]: Currently

File: README.md [hunk 2]
```diff
@@ -5,41 +5,46 @@
...
+* OpenStack/ooi
 
 ## Installation
 
-### Dependencies
+### From binaries
 
-The cloud-provider depends on PyYAML, which is already included as a dependency for
-binary packages and when installing from source.
+Latest packages are made available at [EGI's AppDB](https://appdb.egi.eu/store/software/cloud.info.provider).
```
Comment (gwarf) [inline README.md]: Maybe we could say that all packages are always attached to the releases in GitHub and after they are made available at the AppDB (which is handy for providing a ready-to-use repo).

File: README.md [hunk 3]
```diff
@@ -5,41 +5,46 @@
...
+already include those dependencies (RH based distributions need to enable the
+[EPEL repository](http://fedoraproject.org/wiki/EPEL).
 
-* bdii (available in Ubuntu/Debian repos, for RH based distros it is in EPEL, for
-  Debian wheezy it is available in the backports repo).
-* if you are generating information for OpenStack, you will also need
-  to install python-novaclient.
+For running the provider in a production environment with a BDII you will
+also need the `bdii` package (available in Ubuntu/Debian repos, in EPEL for RH
+based distros it is in EPEL, and in the backports repo for Debian wheezy).
```
Comment (gwarf) [inline README.md]: Wheezy is no more maintained, so probably not worth talking about it

File: README.md [hunk 4]
```diff
@@ -53,26 +58,39 @@ debs:
...
+```
 
-Packages having gone through a the [EGI CMD](https://wiki.egi.eu/wiki/EGI_Cloud_Middleware_Distribution)
-[Stagged Rollout and SQA process](https://wiki.egi.eu/wiki/EGI_Cloud_Middleware_Distribution_process)
-are available in the [CMD repositories](http://repository.egi.eu/).
+##### Provider dependencies
 
-Use the appropriate repository for your distribution and install using the
-OS-specific tools.
+The OpenStack/ooi providers require the following extra libraries:
```
Comment (gwarf) [inline README.md]: Looking at `requirements.txt` it seems that other packages are needed but not listed here (for OS and ON).

File: README.md [hunk 5]
```diff
@@ -137,113 +155,139 @@ cp ../*.deb ~/debs/xenial
...
-installation user should be able to write to it, so it is recommended to create
-it before using pip.
+#### Static information
 
-```sh
-sudo mkdir /etc/cloud-info-provider
-sudo chgrp you_user /etc/cloud-info-provider
-sudo chmod g+rwx /etc/cloud-info-provider
-```
+The cloud-info-provider uses a YAML file describing the static information of
```
Comment (gwarf) [inline README.md]: Could be appropriate to clarify that a default is available in the repo (and/or link to it).

File: README.md [hunk 6]
```diff
@@ -137,113 +155,139 @@ cp ../*.deb ~/debs/xenial
...
 ```sh
-cloud-info-provider-service --yaml-file /etc/cloud-info-provider/static.yaml \
-    --middleware openstack --os-username <username> --os-password <password> \
-    --os-user-domain-name default --os-project-name <tenant> \
-    --os-project-domain-name default --os-auth-url <auth-url>
+cloud-info-provider-service --middleware openstack \
+    --os-auth-type v3oidcaccesstoken \
+    --os-identity-provider egi.eu --os-protocol oidc \
+    --os-access-token $ACCESS_TOKEN \
+    --os-auth-url https://<keystone-endpoint>:5000/v3',
```
Comment (gwarf) [inline README.md]: closing quote (and missing opening one) and coma at EOL

File: README.md [hunk 7]
```diff
@@ -320,17 +366,11 @@ Add your cloud-info-provider to your site-BDII by adding a new URL that looks
...
+ldap://<cloud-info-provider-hostname>:2170/GLUE2GroupID=cloud,o=glue
 ```
 
-## Other deployment modes
-
-**These deployment modes cover special cases that should not be used in
-  production!**
-
-### Running the cloud-provider in a site-BDII
+### GlueSchema2.1
```
Comment (gwarf) [inline README.md]: GLUE 2.1 Schema ?

File: debs/cloud-info-provider-opennebula/debian/control [hunk 8]
```diff
@@ -14,8 +14,7 @@ Section: python
 Depends:
  ${python:Depends},
  ${misc:Depends},
- python-cloud-info-provider (>= 0.9.1),
- python-defusedxml
```
Comment (gwarf) [inline debs/cloud-info-provider-opennebula/debian/control]: Why did you remove defusedxml?

Comment (gwarf) [reply debs/cloud-info-provider-opennebula/debian/control]: OK, it's required by gocdb :)

File: README.md (hunk 4 above)
Comment (gwarf) [reply README.md]: OK, you also updated `requirements.txt`

File: rpm/cloud-info-provider-opennebula.spec [hunk 9]
```diff
@@ -14,7 +14,6 @@ URL: https://github.com/EGI-Foundation/cloud-info-provider
 
 Requires: cloud-info-provider
 # OpenNebula-specific dependencies
-Requires: python-defusedxml
```
Comment (gwarf) [inline rpm/cloud-info-provider-opennebula.spec]: It may be appropriate to still keep the dep here, as it's really required here.

Comment (gwarf) [reply rpm/cloud-info-provider-opennebula.spec]: Maybe my comment is just irrlevant, feel free to ignore :)

Comment (enolfc) [reply rpm/cloud-info-provider-opennebula.spec]: fixed and also added pyOpenSSL and requests that were missing

**Comment by enolfc** [comment]
Changes implemented

File: README.md [hunk 10]
```diff
@@ -5,41 +5,49 @@
...
-that is required for the OpenNebula provider.
+Use the appropriate repository for your distribution and install using the
+OS-specific tools.
 
-Providers-specific dependencies are managed using provider-specific metapackages.
-Those metapackages are dependent on the main cloud-info-provider that is common
-to all the providers and also includes the provider-specific dependencies.
+Dependencies are maintained in the `requirements.txt` file. Binary packages
+already include those dependencies (RH based distributions need to enable the
+[EPEL repository](http://fedoraproject.org/wiki/EPEL).
```
Comment (gwarf) [inline README.md]: Missing closing `)`.

File: README.md [hunk 11]
```diff
@@ -5,41 +5,49 @@
...
+
+For running the provider in a production environment with a BDII you will
+also need the `bdii` package (available in Ubuntu/Debian repos, in EPEL for RH
+based distros it is in EPEL).
+
+Providers-specific dependencies are managed using provider-specific
+metapackages. Those metapackages are dependent on the main cloud-info-provider
+that is common to all the providers and also includes the provider-specific
+dependencies.
```
Comment (gwarf) [inline README.md]: Repetitions across the two sentences.
```
Providers-specific dependencies are managed using provider-specific
metapackages. Those metapackages are dependent on the main cloud-info-provider
that is common to all the providers.
```

File: README.md [hunk 12]
```diff
@@ -5,41 +5,49 @@
...
+* OpenStack
+* OpenStack/ooi
 
 ## Installation
 
-### Dependencies
+### From binaries
 
-The cloud-provider depends on PyYAML, which is already included as a dependency for
-binary packages and when installing from source.
```
Comment (gwarf) [inline README.md]: Double empty lines.

Comment (gwarf) [reply README.md]: cf https://app.codacy.com/app/enolfc/cloud-info-provider/pullRequest?prid=2848700
